sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -83,6 +83,32 @@ impl BlobClient {
         })
     }
 
+    pub fn new_no_credential(
```

Comment:
Need to expose a way to construct a credential-less handwritten abstraction of the clients.

Reply:
- Make this `from_blob_url`
- Should take a full Blob URL, no `container_name` or `blob_name` input
- Parse those elements out, pass along to generated layer
- For now, endpoint can still have query parameters